### PR TITLE
Improve perf of fromCharCode(), fixes #71

### LINKED ISSUE #71: Improve perf of fromCharCode()
Follow-up to #69 and #70; looks like chunking up the array is actually faster: http://jsperf.com/string-fromcharcode-apply-vs-string-fromcharcode-using-/3

### PR DESCRIPTION
**Pull request by nolanlawson**
Cool, that test of ours I mentioned in #70 (the "big png" test) went from 2.7s to 1.4s. Thanks @jessetane for pointing this out!

### CODE REVIEW

**Comment by dcousens** [comment]
@nolanlawson any reason you can't apply this optimization to `binarySlice`?

**Comment by nolanlawson** [comment]
OK, done

File: index.js [hunk 1]
```diff
@@ -622,14 +622,14 @@ function base64Slice (buf, start, end) {
...
 }
 
-function decodeCodePointsArray (array) {
-  // Based on http://stackoverflow.com/a/22747272/680742, the browser with
-  // the lowest argument limit is Chrome, with 0x10000 args.
-  if (array.length < 0x10000) return String.fromCharCode.apply(String, array)
+function decodeCodePointsArray (buf) {
+  var len = buf.length
+
+  if (len <= 0x10000) {
```
Comment (dcousens) [inline index.js]: I might pull this constant out and put the explanation comment on it,  that way its not magically here.

**Comment by dcousens** [comment]
Thanks @nolanlawson

File: index.js (hunk 1 above)
Comment (nolanlawson) [inline index.js]: thanks, that's a good idea!

**Comment by nolanlawson** [comment]
No prob! Any chance of a 3.4.3 release soon? Would unbreak some of our tests in PouchDB.

File: index.js (hunk 1 above)
Comment (dcousens) [inline index.js]: Should just merge all this code into `binarySlice`?
I wonder if the manual string concatenation is slower than `String.fromCharCode.apply(String, buf.slice(start, end))`.
@nolanlawson reckon you could find out?

Comment (nolanlawson) [inline index.js]: The issue isn't manual string concatenation; it's doing an unnecessary `Array.slice()` when the buffer.length is < 0x10000. I think it's pretty clear that doing `slice()` is slower than not doing `slice()`, although a JSPerf would tell us exactly how slow.

Comment (nolanlawson) [inline index.js]: I can indeed verify that slice is slower than no-slice: http://jsperf.com/string-fromcharcode-apply-vs-string-fromcharcode-using-/4

File: index.js [hunk 2]
```diff
@@ -723,13 +723,17 @@ function asciiSlice (buf, start, end) {
 }
 
 function binarySlice (buf, start, end) {
-  var ret = ''
-  end = Math.min(buf.length, end)
```
Comment (dcousens) [inline index.js]: Will re-add this,  shouldn't have been removed. 
Probably need to add a test for it.

Comment (nolanlawson) [inline index.js]: ah, my bad. sorry!

**Pull request merged**